src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -0,0 +1,206 @@
+                settings,
+                ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_IGNORE_USERS,
+                DEFAULT_IGNORED_USERS,
+                false));
+
+        final Set<String> ignoreAuditRequests = ImmutableSet.copyOf(settings.getAsList(
+                ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_REQUESTS,
+                Collections.emptyList()));
+
+        return new AuditConfig(isRestApiAuditEnabled,
+                isTransportAuditEnabled,
+                resolveBulkRequests,
+                logRequestBody,
+                resolveIndices,
+                excludeSensitiveHeaders,
+                ignoredAuditUsers,
+                ignoredComplianceUsersForRead,
+                ignoredComplianceUsersForWrite,
+                ignoreAuditRequests,
+                disabledRestCategories,
+                disabledTransportCategories,
+                opendistrosecurityIndex);
+    }
+
+    private static List<String> getSettingAsList(final Settings settings, final String key, final List<String> defaultList, final boolean ignoreCaseForNone) {
+        final List<String> list = settings.getAsList(key, defaultList);
+        if (list.size() == 1) {
+            final String elem = list.get(0);
+            final String none = "NONE";
+
+            if (ignoreCaseForNone && none.equalsIgnoreCase(elem)) {
+                return Collections.emptyList();
+            }
+            if (!ignoreCaseForNone && none.equals(elem)) {
+                return Collections.emptyList();
+            }
+        }
+        return list;
+    }
+
+    public boolean isRestApiAuditEnabled() {
+        return isRestApiAuditEnabled;
+    }
+
+    public boolean isTransportApiAuditEnabled() {
+        return isTransportAuditEnabled;
+    }
+
+    public boolean shouldResolveBulkRequests() {
+        return resolveBulkRequests;
+    }
+
+    public boolean shouldLogRequestBody() {
+        return logRequestBody;
+    }
+
+    public boolean shouldResolveIndices() {
+        return resolveIndices;
+    }
+
+    public boolean shouldExcludeSensitiveHeaders() {
+        return excludeSensitiveHeaders;
+    }
+
+    public Set<String> getIgnoredAuditUsers() {
+        return ignoredAuditUsers;
+    }
+
+    public Set<String> getIgnoredComplianceUsersForRead() {
+        return ignoredComplianceUsersForRead;
+    }
+
+    public Set<String> getIgnoredComplianceUsersForWrite() {
+        return ignoredComplianceUsersForWrite;
+    }
+
+    public Set<String> getIgnoredAuditRequests() {
+        return ignoreAuditRequests;
+    }
+
+    public EnumSet<AuditCategory> getDisabledRestCategories() {
+        return disabledRestCategories;
+    }
+
+    public EnumSet<AuditCategory> getDisabledTransportCategories() {
+        return disabledTransportCategories;
+    }
+
+    public String getOpendistrosecurityIndex() {
+        return opendistrosecurityIndex;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Instead of `toString()` I recommend delegating logging to `AuditConfig` class itself: `public void log(Logger log)`.

Reply:
Maybe make a new issue to add this pattern to other classes as well? 

I feel the correct way of doing this is make a new interface Loggable for log and then implement it. Also should be able to configure the log level.

Reply:
I don't see a use case for `Loggable` interface. I think that INFO is the most appropriate level for logging AuditConfig. For now, I would not extend the pattern to other classes.